XLS-0100-smart-escrows/README.md

```diff
@@ -0,0 +1,552 @@
+- **Achievement Rewards**: Unlock funds upon the completion of a certification or milestone.
+- **Closed Airdrops**: Enable token receipt only for holders of specific NFTs.
+- **P2P Bets**: Enable value transfer based on oracle data, such as whether XRP’s price reaches a target value on a given date.
+- **"Options" Contracts**: Condition release based on price oracles or DEX prices for derivative or options-like structures.
+- **Token Swaps**: Trustless token swaps where one escrow releases funds only once another escrow is created.
+- **NFT-Based Releases**: Condition the release of funds on the transfer or sale of a particular NFT.
+- **Token Vesting**: Lock assets for investors, allowing gradual or milestone-based release.
+- **Cascading Escrows**: Link escrows in sequences where one escrow depends on the completion of another (e.g., real estate transactions).
+- **Auction Mechanisms**: Lock funds for auction participants, with conditions for finalization based on bid outcomes.
+- **Treasury Management**: Structure conditions for internal treasury and custodial operations.
+
+## 3. Overview
+
+The design involves a minimal, programmable code block attached to an Escrow object that executes logic to determine whether the Escrow can be completed. This follows the model of a restricted, on-ledger “smart contract,” where conditions can be coded and verified directly on-chain. This is an incremental extension of XRPL's CryptoConditions, enhancing flexibility with custom validation while maintaining computational limits and simplicity.
+
+The code block is attached to the Escrow on creation (in an `EscrowCreate` transaction) and executed on an `EscrowFinish` transaction for that escrow.
+
+This design is essentially an extension of the idea of [using an Escrow as a Smart Contract](https://xrpl.org/docs/tutorials/how-tos/use-specialized-payment-types/use-escrows/use-an-escrow-as-a-smart-contract#use-an-escrow-as-a-smart-contract).
+
+The details of the WASM engine and the API are in a separate XLS.
+
+This proposal involves:
+
+- Modifications to the `Escrow` and `FeeSettings` ledger objects
+- Modifications to the `EscrowCreate`, `EscrowFinish`, and `SetFee` transactions
+
+This feature will require an amendment, tentatively titled `SmartEscrow`.
+
+### 3.1. Background: XRPL Escrows
+
+There are currently two ways to release an escrow: **Time-based** and **Condition-based.**
+
+A **time-based** escrow releases its funds after a certain specified time. In more technical terms, a time-based escrow has the `FinishAfter` field and can be released anytime after the time specified in the `FinishAfter` field.
+
+A **condition-based** escrow essentially only releases its funds if a password is provided. In more technical terms, a condition-based escrow has the `Condition` field and can be released if the matching `Fulfillment` field (essentially the reverse of a hash) is provided on the `EscrowFinish` transaction.
+
+#### 3.1.1. Table of Allowed Options
```

Comment:
Consider removing this table and simply linking to xrpl.org docs. E.g., https://xrpl.org/docs/references/protocol/transactions/types/escrowcreate#escrowcreate-fields

Some rationale:
1. This mirror what you're doing on line 75, and prevents duplicating functional documentation
2. You have a new table below that lists all of this information again (including the new rows for SE).